### PR TITLE
HDDS-7726. EC: Enhance datanode reconstruction log message

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Include more information in messages from `ECReconstructionCoordinatorTask`.  Source/target datanodes are identified by UUID.

To facilitate the change, move some logic from `ECReconstructionCoordinatorTask` to `ECReconstructionCommandInfo`.

https://issues.apache.org/jira/browse/HDDS-7726

## How was this patch tested?

```
$ mvn -Dtest=TestECContainerRecovery clean test
...

$ grep ECReconstructionCommand hadoop-ozone/integration-test/target/surefire-reports/org.apache.hadoop.ozone.container.TestECContainerRecovery-output.txt
... INFO  reconstruction.ECReconstructionCoordinatorTask (ECReconstructionCoordinatorTask.java:run(97)) - Completed ECReconstructionCommand{containerID=1, replication=rs-3-2-1024, missingIndexes=[1], sources={2:dcaef49d-a60e-42a7-91d8-740a95ae2403,3:fa4feb1a-bb51-4c8b-9a79-ba0dae67a626,4:0af75ffe-7d73-4d6f-859d-14da589f77ba,5:770b81eb-e1a4-478e-83ea-2cb8ce8876f1}, targets={1:ca9c6427-8e00-426f-a28b-138bf9583c8e}} in 241 ms
... WARN  reconstruction.ECReconstructionCoordinatorTask (ECReconstructionCoordinatorTask.java:run(99)) - Failed ECReconstructionCommand{containerID=2, replication=rs-3-2-1024, missingIndexes=[1], sources={2:a9005b92-4153-4535-b3c2-6c1d774e586a,3:b29d4ce4-0416-4ff7-afb9-b8968068d7ab,4:edacb8d4-c8b9-424e-8b1c-f2c9be0858a3,5:4cb977e9-d7ed-4833-92b5-c1181ce1db2e}, targets={1:0af75ffe-7d73-4d6f-859d-14da589f77ba}}
```

Regular CI:
https://github.com/adoroszlai/hadoop-ozone/actions/runs/3861802806

### CODE REVIEW

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ec/reconstruction/ECReconstructionCoordinatorTask.java [hunk 1]
```diff
@@ -88,40 +84,26 @@ public void run() {
...
+          reconstructionCommandInfo.getEcReplicationConfig(),
+          reconstructionCommandInfo.getSourceNodeMap(),
+          reconstructionCommandInfo.getTargetNodeMap());
+      long elapsed = Time.monotonicNow() - start;
+      LOG.info("Completed {} in {} ms", reconstructionCommandInfo, elapsed);
     } catch (IOException e) {
-      LOG.warn(
-          "Failed to complete the reconstruction task for the container: "
-              + reconstructionCommandInfo.getContainerID(), e);
+      LOG.warn("Failed {}", reconstructionCommandInfo, e);
```
Comment (sodonnel) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ec/reconstruction/ECReconstructionCoordinatorTask.java:99]: Might be worth adding the elapsed time to the fail message too, incase it is failing after some long delay or timeout?

**Comment by sodonnel** [comment]
This change looks basically good. I just had one suggestion about adding the elapsed time to the fail message too.

I also wonder about logging the DN UUID. Logging DatanodeDetails.toString() is really too verbose and make the logs quite long, so its a good idea to avoid it. In #4153 I went with logging "DN host / IP" to make it less verbose. If someone is debugging an issue, it might be easier to work with that rather than a UUID, as it saves another lookup to find the host you may want to look at, but it could be argued either way I think. I am fine to leave it as it is I think.

**Comment by adoroszlai** [comment]
> I went with logging "DN host / IP" to make it less verbose. If someone is debugging an issue, it might be easier to work with that rather than a UUID, as it saves another lookup to find the host you may want to look at, but it could be argued either way I think. I am fine to leave it as it is I think.

In normal usage hostname is better.  UUID might be better for integration tests with mini cluster, where all nodes are on the same host.  I guess I'll go with your solution, and maybe we can add a test-specific check later.

**Review by sodonnel — APPROVED** [review]
LGTM

**Comment by adoroszlai** [comment]
Thanks @sodonnel for reviewing and merging this.

**Pull request merged**